fhttp/http_client.go

```diff
@@ -145,6 +147,24 @@ func (h *HTTPOptions) ResetHeaders() {
 func (h *HTTPOptions) InitHeaders() {
 	h.ResetHeaders()
 	h.extraHeaders.Add("User-Agent", userAgent)
+	err := h.ValidateAndAddBasicAuthentication()
```

Comment:
I'm not sure this is the right place for this, the init is the baseline for the headers
(but then again the whole header part isn't super clean at the moment)

Reply:
I put header in this method because it solves the fast client as well.  We can move the method, if the header part becomes clean in the future.

Reply:
I should have insisted... see #282